### PR TITLE
DEV-307  + DEV-308 - support contributor_name and transferring collections

### PR DESCRIPTION
**Pull request by respinos**
- support fetching/editing `contributor_name`
- ensure that incoming metadata values are plaintext
- add `transfer_collection` to transfer owner

### CODE REVIEW

File: Collection.pm [hunk 1]
```diff
@@ -841,6 +854,21 @@ sub _edit_metadata {
...
 
+    if ( $value ) {
+        # sanitize the input
+        require XML::LibXML;
+        require HTML::Entities;
+        my $html_parser = XML::LibXML->new();
+        eval {
+            $value = HTML::Entities::decode_entities($value);
+            my $html_dom = $html_parser->parse_html_string("<div>$value</div>", { recover => 2 });
+            $value = $html_dom->textContent;
```
Comment (aelkiss) [inline Collection.pm:865]: Are we sanitizing HTML anywhere else in `mdp-lib`? Is there any trusted standard way of doing that in perl? This definitely seems like a fraught area.

In fact I think there is still an XSS vulnerability here -- if you double-escape entities and do e.g.

```
&amp;lt;script&amp;gt; alert("got you"); &amp;lt;/script&amp;gt;
```
then the first `decode_entities` decodes it to

```
&lt;script&gt; alert("got you"); &lt;/script&gt;
```

and `textContent` appears to decode it again to

```
<script> alert("got you"); </script>
```

I wasn't able to exploit this on beta-1, but I'm not sure why -- with double-escaping it got back just the inner text (`alert("got you")`) and with triple-escaping I got it back singly-escaped (`&lt;script&gt; alert("got you"); &lt;/script&gt;`)

So, I *think* we're good, but I can't explain why just from this.

Comment (respinos) [reply Collection.pm:865]: I could not find any modern sanitizing modules 

The collection edit widgets builds its own form to submit, and that seems to remove much of the XSS threat. The only way HTML has gotten into the collection descriptions has been via the batch edit script.

The updated XSLT uses `xsl:value-of` on these values --- I'm sure there's a way to trip this! But forcing any of these into the database doesn't result in HTML out.

* `&amp;lt;u&amp;gt;This is Fine&amp;lt;/u&amp;gt;`
* `&lt;u&gt;This is Fine&lt;/u&gt;`
* `<u>This is Fine</u>`

Comment (aelkiss) [reply Collection.pm:865]: :+1:

**Review by aelkiss — APPROVED** [review]
This all seems straightforward enough; I guess my only question is why some of these methods are here rather than in `mb` -- are they used by other `mdp-lib` apps besides `mb`?

**Comment by respinos** [comment]
`pt` and `ls` use the `Collection` and `CollectionSet` classes, so they're in `mdp-lib`. The CRUD methods have always been here.

**Pull request merged**